assets/javascripts/kitten/components/form/radio-button.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import classNames from 'classnames'
+
+export class RadioButton extends React.Component {
+  renderContent() {
+    if (!this.props.content)
+      return
+
+    const labelContentsClassNames = classNames(
+      'k-RadioButton__labelContents',
+      { 'k-RadioButton__labelContents--large': this.props.contentLarge }
+    )
+    return (
+      <div className={ labelContentsClassNames }>
+        <p>{ this.props.content }</p>
+      </div>
+    )
+  }
+
+  render() {
+    let radioButtonClassNames = classNames (
+      'k-RadioButton__label',
+      { 'k-RadioButton__label--large': this.props.large },
+    )
+
+    return (
+      <div className="k-RadioButton">
+        <input id={ this.props.id }
+               type="radio"
+               name={ this.props.name }
+               className="k-RadioButton__input"
+               value={ this.props.value }
+               defaultChecked={ this.props.checked }
+               disabled={ this.props.disabled } />
```

Comment:
Ici on pourrait profiter d'une des possibilités de React, c'est de passer toutes les props restantes à l'input. Ça permettrait à l'avenir de pouvoir passer des attributs différents (`data-*`, `aria-*`, `required`, …) à notre input sans avoir à toutes les lister dans notre composant.

On peut faire ça en utilisant le "spread" `{ ...this.props }` :

```js
return <input id={ this.props.id }
              type="radio"
              name={ this.props.name }
              className="k-RadioButton__input"
              value={ this.props.value }
              checked={ this.props.checked }
              defaultChecked={ this.props.defaultChecked }
              { ...this.props } />
```

En faisant ça on peut retirer les lignes où on ne faisant que repasser une propriété du même nom :

```js
return <input type="radio"
              className="k-RadioButton__input"
              { ...this.props } />
```

Si on veut éviter de passer les propriétés qu'on utilise dans notre composant mais qu'on ne veut pas mettre sur un input, on peut faire ça :

```js
const { large, text, content, contentLarge, ...others } = this.props
return <input type="radio"
              className="k-RadioButton__input"
              { ...others } />
```